test/autorest/generated/headergroup/internal/headergroup/models.go

```diff
@@ -39,175 +41,189 @@ func (e Error) Error() string {
 type HeaderParamStringResponse struct {
-	// StatusCode contains the HTTP status code.
-	StatusCode int
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
 }
 
 // HeaderResponseBoolResponse contains the response from method Header.ResponseBool.
 type HeaderResponseBoolResponse struct {
-	// StatusCode contains the HTTP status code.
-	StatusCode int
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
+	Value *bool
 }
 
 // HeaderResponseByteResponse contains the response from method Header.ResponseByte.
 type HeaderResponseByteResponse struct {
-	// StatusCode contains the HTTP status code.
-	StatusCode int
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
+	Value *[]byte
 }
 
 // HeaderResponseDateResponse contains the response from method Header.ResponseDate.
 type HeaderResponseDateResponse struct {
-	// StatusCode contains the HTTP status code.
-	StatusCode int
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
+	Value *time.Time
 }
 
 // HeaderResponseDatetimeRFC1123Response contains the response from method Header.ResponseDatetimeRFC1123.
 type HeaderResponseDatetimeRFC1123Response struct {
-	// StatusCode contains the HTTP status code.
-	StatusCode int
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
+	Value *time.Time
 }
 
 // HeaderResponseDatetimeResponse contains the response from method Header.ResponseDatetime.
 type HeaderResponseDatetimeResponse struct {
-	// StatusCode contains the HTTP status code.
-	StatusCode int
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
+	Value *time.Time
 }
 
 // HeaderResponseDoubleResponse contains the response from method Header.ResponseDouble.
 type HeaderResponseDoubleResponse struct {
-	// StatusCode contains the HTTP status code.
-	StatusCode int
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
+	Value *float64
 }
 
 // HeaderResponseDurationResponse contains the response from method Header.ResponseDuration.
 type HeaderResponseDurationResponse struct {
-	// StatusCode contains the HTTP status code.
-	StatusCode int
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
+	Value *time.Duration
 }
 
 // HeaderResponseEnumResponse contains the response from method Header.ResponseEnum.
 type HeaderResponseEnumResponse struct {
-	// StatusCode contains the HTTP status code.
-	StatusCode int
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
+	Value *GreyscaleColors
 }
 
 // HeaderResponseExistingKeyResponse contains the response from method Header.ResponseExistingKey.
 type HeaderResponseExistingKeyResponse struct {
-	// StatusCode contains the HTTP status code.
-	StatusCode int
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
+	UserAgent *string
 }
```

Comment:
You will also want to add any description text as a comment above the header value.  If the description is empty we can put some more generic text like "UserAgent contains the value from the User-Agent header response." or something like that.

Reply:
And include a new-line between entries.

Reply:
Ok done.